symphony/lib/boot/func.utilities.php

```diff
@@ -8,32 +8,32 @@
  * Redirects the browser to a specified location. Safer than using a
  * direct header() call
  *
- *  @param string $url
+ *  @param string $location
+ *  @param string $status (optional)
  */
-function redirect ($url)
+function redirect($location, $status = '302 Found')
 {
-    // Just make sure.
-    $url = str_replace('Location:', null, $url);
-
     if (headers_sent($filename, $line)) {
-        echo "<h1>Error: Cannot redirect to <a href=\"$url\">$url</a></h1><p>Output has already started in $filename on line $line</p>";
-        exit;
-    }
 
-    // convert idn back to ascii for redirect
+        // throw exception if headers already sent
```

Comment:
Just to make sure I got it right, you prefer

    if (headers_sent($filename, $line)) {
        // throw exception if headers already sent
        throw new SymphonyErrorPage(sprintf(
            'Cannot redirect to <a href="%s">%s</a>. Output has already started in %s on line %s.',
            $location, $location, $filename, $line
        ));
    }

    if (function_exists('idn_to_ascii')) {
        // convert idn back to ascii for redirect
        $location = str_replace(HTTP_HOST, idn_to_ascii(HTTP_HOST), $location);
    }

over

    if (headers_sent($filename, $line)) {

        // throw exception if headers already sent

        throw new SymphonyErrorPage(sprintf(

            'Cannot redirect to <a href="%s">%s</a>. Output has already started in %s on line %s.',

            $location, $location, $filename, $line
        ));
    }

    if (function_exists('idn_to_ascii')) {

        // convert idn back to ascii for redirect

        $location = str_replace(HTTP_HOST, idn_to_ascii(HTTP_HOST), $location);
    }

in terms of readability?